metrics.nim

```diff
@@ -166,6 +166,12 @@ when defined(metrics):
       result = result !& label.hash
     result = !$result
 
+  # Hash-based data types like OrderedSet don't just compare hashes to determine
+  # if a key is present, but also compare the keys themselves, so we need to
+  # override the `==` operator.
+  method `==`*(x, y: Collector): bool {.base.} =
+    x.hash == y.hash and x.name == y.name and x.labels == y.labels
```

Comment:
no need to compare the hash though since we're comparing all the bits of the fields that make up the hash ;)

Reply:
It makes the likeliest branch faster ;-)

Reply:
the hash table impl has already made that check - in general, comparing things and computing their hash is about the same, both iterate over the bytes

Reply:
Right, I'll remove it.

Reply:
if you want to make things more efficient, you'll want to avoid the 5 lookups performed for every metric update:

* 2 here: https://github.com/status-im/nim-metrics/blob/7349a421cff02c27df9f7622801459d6b5d65dcd/metrics.nim#L290
* 2 here: https://github.com/status-im/nim-metrics/blob/7349a421cff02c27df9f7622801459d6b5d65dcd/metrics.nim#L358
* 1 here: https://github.com/status-im/nim-metrics/blob/7349a421cff02c27df9f7622801459d6b5d65dcd/metrics.nim#L361

each of these does a full lookup in the hash table meaning both hash and equals, potentially multiple times depending on the composition of the table - copying the label values every time also takes up time. the former can be addressed with an `update` proc that takes a `var` parameter to which you pass the looked-up metric - the latter by taking `seq[string]` as input (the copy actually shows up in profiling)